src/Controls/src/Core/DispatcherExtensions.cs

```diff
@@ -22,11 +22,17 @@ public static IDispatcher FindDispatcher(this BindableObject? bindableObject)
 			if (Dispatcher.GetForCurrentThread() is IDispatcher globalDispatcher)
 				return globalDispatcher;
 
-			// If BO is of type Application then check for its Dispatcher
+			// If BO is of type Application then return the Dispatcher from ApplicationDispatcher
 			if (bindableObject is Application app &&
-				app.FindMauiContext() is IMauiContext appMauiContext &&
-				appMauiContext.Services.GetService<IDispatcher>() is IDispatcher appHandlerDispatcher)
-				return appHandlerDispatcher;
+				app.FindMauiContext() is IMauiContext appMauiContext)
+			{
+				if (appMauiContext.Services.GetService<ApplicationDispatcher>()?.Dispatcher is IDispatcher appDispatcherServiceDispatcher)
+					return appDispatcherServiceDispatcher;
```

Comment:
```suggestion
				if (appMauiContext.Services.GetOptionalApplicationDispatcher() is IDispatcher appDispatcherServiceDispatcher)
					return appDispatcherServiceDispatcher;
```

The app dispatcher is overrideable by using keyed services, so we need to make sure we check that too. The GetOptionalApplicationDispatcher method does this.

Reply:
Just to verify, someone can register a service keyed to `IApplication` as a singleton but then `IDispatcher` non-keyed will still be a scoped correct?

Reply:
yes

Reply:
I do that for the tests to override the app dispatcher so I don't need to fiddle with the AppDispatcher things publicly.